plugins/shapefile/main.py

```diff
@@ -32,34 +35,16 @@ def load_data(self):
             self.metadata = shp.meta
             projection = Proj(init=self.metadata['crs']['init'])
 
-            # Determine extent
-            def explode(coords):
-                """ Borrowed from sgilles - https://gis.stackexchange.com/questions/90553/fiona-get-each-feature-extent-bounds
-                Explode a GeoJSON geometry's coordinates object and yield coordinate tuples.
-                As long as the input is conforming, the type of the geometry doesn't matter."""
-                for e in coords:
-                    if isinstance(e, (float, int)):
-                        yield coords
-                        break
-                    else:
-                        for inner_coords in explode(e):
-                            yield inner_coords
-
-            def bbox(feature):
-                x, y = zip(*list(explode(feature['geometry']['coordinates'])))
-                return min(x), min(y), max(x), max(y)
-
-            xmin, ymin, xmax, ymax = [None] * 4
-
-            for f in shp:
-                if xmin is None:
-                    xmin, ymin, xmax, ymax = bbox(f)
-                else:
-                    _xmin, _ymin, _xmax, _ymax = bbox(f)
-                    xmin, ymin, xmax, ymax = \
-                        min([xmin, _xmin]), min([ymin, _ymin]), max([xmax, _xmax]), max([ymax, _ymax])
+        driver = ogr.GetDriverByName('ESRI Shapefile')
```

Comment:
`fiona` doesn't have (as far as I have found) a way to determine the number of features directly. GDAL also has *very* convenient functions for accessing the spatial extent and number of features, and `fiona` does not.